lib/audit/catalog_to_moab.rb

```diff
@@ -5,19 +5,21 @@
 class CatalogToMoab
 
   # allows for sharding/parallelization by storage_dir
-  def self.check_version_on_dir(last_checked_b4_date, storage_dir)
+  def self.check_version_on_dir_of_batch(last_checked_b4_date, storage_dir, limit)
     # TODO: ensure last_checked_version_b4_date is in the right format for query - see #485
-    pcs = PreservedCopy
-          .joins(:endpoint)
-          .where(endpoints: { storage_location: storage_dir })
-          .where('last_version_audit IS NULL or last_version_audit < ?', last_checked_b4_date)
-          .order('last_version_audit IS NOT NULL, last_version_audit ASC')
-    pcs.find_each do |pc|
+    pcs = PreservedCopy.least_recent_version_audit(last_checked_b4_date, storage_dir).limit(limit)
+    pcs.each do |pc|
       c2m = CatalogToMoab.new(pc, storage_dir)
       c2m.check_catalog_version
     end
   end
 
+  def self.check_version_on_dir(last_checked_b4_date, storage_dir)
+    unless PreservedCopy.least_recent_version_audit(last_checked_b4_date, storage_dir).count.zero?
```

Comment:
question:  is there a way to avoid doing this query twice per batch (once here in `check_version_on_dir` and once in `check_version_on_dir_of_batch`) ?   I'm fine with it being a second ticket.

Reply:
yeah, we could probably do that without too much trouble.  though it's also worth noting that a count query like this should usually be much cheaper to run than the same query actually fetching the records`*`.  or so DB devs better than me have told me in the past.  i think the query optimizer has tricks for shortcutting unnecessary computations/retrievals, plus it's just an integer value getting sent back from the DB instead of 1000 rows, plus those 1000 rows don't have to get turned into 1000 ActiveRecord objects in memory.

`*` and so if you really only wanted the count, doing this with `#size` instead of `#count` wouldn't be advisable because you'd be getting the records and checking array length after a bunch of AR instantiations instead of doing a count query in PG.

Reply:
you could also just do it without getting the count, i think?  if there aren't any rows that satisfy the query, the array is empty and nothing happens.

i think the way to use count would be to loop until there's nothing left that's older than the cut-off.  something like:
```ruby
while PreservedCopy.least_recent_version_audit(last_checked_b4_date, storage_dir).count > 0
  check_version_on_dir_of_batch(last_checked_b4_date, storage_dir, Settings.c2m_sql_limit)
end
```

alternatively, i just found out ruby has an `until`, so you could also do `until PreservedCopy.least_recent_version_audit(last_checked_b4_date, storage_dir).count.zero?`.